odbc-api/src/environment.rs

```diff
@@ -17,6 +17,9 @@ use widestring::{U16CStr, U16Str, U16String};
 /// manager. There must only be one environment in the entire process.
 pub struct Environment {
     environment: handles::Environment,
+    /// Locks the drivers to enforce that only a single thread is able to query driver information
```

Comment:
How about: "Used to synchronize access to the iterator over driver and data source information"?

Reply:
Sounds good!